projects/plugins/jetpack/extensions/blocks/mailchimp/mailchimp.php

```diff
@@ -264,3 +271,85 @@ function render_deprecated_button( $attr ) {
 		wp_kses_post( $text )
 	);
 }
+
+/**
+ * Registers the settings to manage the Mailchimp connection.
+ */
+function register_admin_settings() {
+	Assets::register_script(
+		'jetpack-mailchimp-admin-extra-settings',
+		Jetpack_Gutenberg::get_blocks_directory() . '/mailchimp/admin.js',
+		JETPACK__PLUGIN_FILE,
+		array(
+			'textdomain' => 'jetpack',
+		)
+	);
+
+	External_Connections::add_settings_for_service(
+		'writing',
+		array(
+			'service'      => 'mailchimp',
+			'title'        => __( 'Mailchimp', 'jetpack' ),
+			'signup_link'  => 'https://public-api.wordpress.com/rest/v1.1/sharing/mailchimp/signup',
+			'description'  => __( 'Allow users to sign up to your Mailchimp mailing list.', 'jetpack' ),
+			'script'       => 'jetpack-mailchimp-admin-extra-settings',
+			'support_link' => array(
+				'wpcom'   => 'https://wordpress.com/support/wordpress-editor/blocks/mailchimp-block/',
+				'jetpack' => 'mailchimp-block',
+			),
+		)
+	);
+
+	add_action( 'load-options.php', __NAMESPACE__ . '\update_settings' );
+}
+
+/**
+ * Update the site options that are related to Mailchimp.
+ */
+function update_settings() {
+	$action      = ! empty( $_REQUEST['action'] ) ? sanitize_text_field( wp_unslash( $_REQUEST['action'] ) ) : '';
+	$option_page = ! empty( $_REQUEST['option_page'] ) ? sanitize_text_field( wp_unslash( $_REQUEST['option_page'] ) ) : '';
+	$audience    = ! empty( $_REQUEST['jetpack-mailchimp-audience'] ) ? sanitize_text_field( wp_unslash( $_REQUEST['jetpack-mailchimp-audience'] ) ) : '';
```

Comment:
Q: Shouldn't this default to 'none' instead of an empty string? 🤔 This way, you won't need to check if $audience is an empty string.

Reply:
No, because then a request to `options.php?action=update&option_page=wiriting` without the `jetpack-mailchimp-audience` query param would reset the Mailchimp settings. We only want to reset them when there is an explicity `jetpack-mailchimp-audience=none` param.